tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLHighTier.java

```diff
@@ -77,8 +78,8 @@ public OCLHighTier(OptionValues options, TornadoDeviceContext deviceContext, Can
                 appendPhase(new IterativeConditionalEliminationPhase(canonicalizer, false));
             }
         }
-
         appendPhase(new TornadoTaskSpecialisation(canonicalizer));
+        appendPhase(new TornadoBatchGlobalIndexOffset());
```

Comment:
You mentioned that this phase is batch-specific. However, it is invoked also when batch processing is not used. Right?